testing/python_lib/test_fot.py

```diff
@@ -639,7 +645,11 @@ def test_dhcp_reflection(self):
         # Test (lack of) DHCP reflection for operational device
         device_tcpdump_text, vlan_tcpdump_text = self._internal_dhcp('forch-faux-4')
         self.assertTrue(re.search("DHCP.*Reply", device_tcpdump_text))
-        self.assertFalse(re.search("DHCP.*Reply", vlan_tcpdump_text))
+        allowed_vlan = '171'
+        vlans = re.findall(r'(?<=vlan )\w+', vlan_tcpdump_text)
+        no_vlan = not re.search("DHCP.*Reply", vlan_tcpdump_text)
+        is_vlan_allowed = all(vlan == allowed_vlan for vlan in vlans)
+        self.assertTrue(no_vlan or is_vlan_allowed)
```

Comment:
So this test (assertTrue) should be replicated in the other case (forch-faux-1) to make sure that it's false (so same logic, but with assertTrue at the end). I.e. extract to a function (_check_vlans) and then have assertTur(check_vlans()) and assertFalse(checkVlans().  Otherwise it's too hard to tell that, e.g., the code you have just doesn't always return True for some reason.